### DISCUSSION TITLE
Make Inviter/Invited reward visible to users

### DISCUSSION THREAD

**Discussion by muguika** [cambiatus/frontend#804] [Ideas]
**What we have**
We have a feature that allows members to invite others. The community admin can define a reward for the member that invites a new person to join the community and to new members that join through the invite link

**Proposal**
Today the reward is not visible to members. Add it to the app so members are aware of it and motivated to use it.

**Why**
The reward is not displayed in the app, nor communicated in any way to members

**How**
It could be added to the invite link popup here. 
![image](https://user-images.githubusercontent.com/25396167/182745842-3b21dded-bffb-467b-9660-baa062e048c6.png)
If the community has defined a inviter/invited reward, the text could change to: 

"Invite your friends and earn (currency name)"
"SEND THIS URL TO YOUR FRIENDS AND EARN (inviter reward + currency symbol) when they join"

If the community hasn't defined a inviter/invited reward, the text could remain the same as today.

**Additional context**
I noticed this today during the demo and co-design session with CofiBlocks, a new community joining Cambiatus.

**Comment by lucca65** [comment]
This is good, we could improve this screen by a lot!

We already have plenty of information on our databases to provide the following with little change to our API:

- Number of people invited
- Amount of tokens earned in the invitation
- Preview of this link on social networks (like we have on the category form)

With some more work we could also do:

- Provide a picture of the community on the link
- Add the ability to add a custom text to the invite
- Give insights to inviters on how well their link is going
- Allowing inviter to edit the link. Now is a random string

For the admins/communities we could also provide insights:

- How many links were created
- How many people joined via links
- Top inviters/links

And just to back it up on this. Play4Change actually asked for this a while ago:

as they have an invitation action/objective and would like to track *how* people join the community: is it via invite link? maybe just by accessing it?

They also have a different ruling for their Governance token. They wish it to be invite only (makes sense right) and wanted new members to be approved. I think this is a very much necessary theme moving forward into collective decision making and would be awesome to investigate further.

A lot of the actions I've suggested here are super simple, others not so much, but all of them pretty straightforward

**Reply by henriquecbuss** [reply]
Play4Change also asked for a very interesting thing, and I think it's not too hard to implement: they want to have some special invite links that, when a user joins through them, they automatically gain some specific roles.

For example, they could create a link for players and one for coaches. Then they don't need to manually check who's new on the community and manually give them the correct roles

**Reply by lucca65** [reply]
+1 this makes a lot of sense, different invites would make the community grow organized. 

I can think of an example that uses this for CMBGO.

We can have a role like `community-leaders` that is auto assigned when we share an invite like `cmbgo.cambiatus.io/i/community-leaders`. This would auto assign the role. 

We now can have claims that can only be voted by that role. 🤯 

Automatically scoping users and making actions that are relevant to them!

**Reply by muguika** [reply]
All this sounds fantastic folks! Especially excited about the personalized links that could lead to roles and then to specific claims for that role 👏

**Comment by lucca65** [comment]
@muguika I'm converting this to a discussion, we can derive the issues from there. I'm guessing it will become at least 2 issues based only on what you've suggested: one for @cambiatus/designers and one for @cambiatus/devs

**Comment by juramos-2020** [comment]
Boa @lucca65 !
Lembro de ja termos discutido essa ideia de melhorar o pop-up de convite e agora, com esse tipo de discursão, podemos levantar e documentar algumas ideias.

Vejo muito potencial nessa melhoria, principalmente, como uma ferramenta administrativa que vai ajudar os lideres das comunidades a analisar alguns comportamentos.

**Reply by lucca65** [reply]
@juramos-2020 a gente chegou a criar alguma coisa disso? Issue, conversa do Slack, que você lembre?

**Reply by juramos-2020** [reply]
Acho que falamos em algum rice apenas. Ainda não temos nada desenhado.
